knotx-adapter/knotx-adapter-common/src/main/java/io/knotx/adapter/common/http/HttpClientFacade.java

```diff
@@ -49,23 +51,44 @@
 
   private final WebClient webClient;
 
-  public HttpClientFacade(WebClient webClient, List<ServiceMetadata> services) {
+  private final JsonObject customRequestHeader;
+
+  public HttpClientFacade(WebClient webClient, HttpAdapterConfiguration configuration) {
     this.webClient = webClient;
-    this.services = services;
+    this.services = configuration.getServices();
+    this.customRequestHeader = configuration.getCustomRequestHeader();
   }
 
   public Single<ClientResponse> process(AdapterRequest message, HttpMethod method) {
     return Single.just(message)
         .doOnSuccess(this::validateContract)
         .map(this::prepareRequestData)
-        .flatMap(serviceRequest -> callService(serviceRequest, method))
+        .flatMap(
+            serviceRequest -> callService(serviceRequest, method)
+                .doOnSuccess(resp -> logResponse(serviceRequest, resp)))
         .flatMap(this::wrapResponse);
   }
 
+  private void logResponse(Pair<ClientRequest, ServiceMetadata> request,
+      HttpResponse<Buffer> resp) {
+    if (resp.statusCode() != HttpResponseStatus.OK.code()) {
```

Comment:
Not sure if we should `WARN` about other status codes than `200`.
Statuses other than `200` are sometimes used to communicate status of REST operation (e.g. like `201`, `203` or `204`).
Maybe this should be `DEBUG` level here or `WARN` only `4XX` and `5XX` responses and `TRACE` others?

Reply:
Agree.